index.js

```diff
@@ -172,11 +195,19 @@ class PearIPC extends ReadyResource {
     this.server.on('connection', async (stream) => {
       const client = new this.constructor({ ...this._opts, stream })
       client.id = this._clients.alloc(client)
-      stream.once('end', () => { client.close().finally(() => client.emit('close')) })
+      stream.once('end', () => { client.close() })
```

Comment:
Not related to this PR, but I think client.close() can result in an unhandled rejection (assuming it runs the _close method in this index.js file). Maybe we can add safety-catch?

Reply:
close never throws, the code relies on this, so while it returns a promise, it is non-dangerous

I'll add a comment to the close method to make this clear

Reply:
Okay, makes sense. Note that you do catch the `close` of the clients on line 299, so probably can remove that one too then

Reply:
true! removed